cognite/dm_clients/domain_modeling/domain_model_api.py

```diff
@@ -92,6 +94,11 @@ def apply(self, items: Iterable[DomainModelT], ext_id_prefix: str = "") -> List[
           3. Create nodes.
           4. Create edges for those one-to-many relationships.
         """
+        if ref_items := [item for item in items if item._reference]:
+            raise ValueError(
+                f"References passed into {type(self).__name__}.apply(): {[item.externalId for item in ref_items]}"
```

Comment:
Can this error message give more help to the user? What should they do instead?

Reply:
Yeah, I was thinking about listing possible attributes (as in "Movie doesn't have attr. foobar, it has: title, release, actors, director..."), but then I thought this _might_ be considered as a security leak. Maybe. WDYT?

Reply:
🤔 But also, if you do something like `movie.foobar` you get just a regular old `AttributeError`. This is similar. If we "upgrade" this to be more informative, we would be in a strange situation where `a_movie.foobar` gives a bare-bones exception, but `client.movie.connect.foobar` gives a rich one.

Reply:
I think attribute error is good enough, all good then.